### DISCUSSION TITLE
Removing support for -uu ?

### DISCUSSION THREAD

**Discussion by vpax** [zeek/zeek#1728] [Archive]
Back in February as part of the script optimization work I added a `-u` flag for reporting potential usage issues, including static analysis for finding variables that might be used without having first been defined.  The changes included support for `-uu` which turned on additional (expensive) such analysis for record fields as well.

Per the comment closing out https://github.com/zeek/zeek/pull/1708, I've reworked the internals associated with tracking variable usage, for a major speedup.  The new code supports `-u` but not `-uu`.  I'm thinking maybe we just get rid of `-uu` for now; doing so will allow us to remove 2,500 LOC, namely the old way of tracking variable usage, now only needed for `-uu`.

It should be possible to support `-uu` using the new formulation, but it'll take some work on my part that I'd rather apply to more pressing script optimization to-dos.

So: how about we remove `-uu` for now?

**Comment by timwoj** [comment]
`-uu` was added during the 4.1 time frame right? If so, I think we can just take it out without any real ramifications.

**Reply by vpax** [reply]
Yep - just to be sure, I downloaded 4.0.3 and confirmed that there's no `-u` option.

**Comment by timwoj** [comment]
This was resolved via https://github.com/zeek/zeek/pull/1794.
